src/core/Akka.Streams/Implementation/IO/OutputStreamSourceStage.cs

```diff
@@ -126,7 +126,6 @@ private void OnDownstreamFinish()
             {
                 //assuming there can be no further in messages
                 _downstreamStatus.Value = Canceled.Instance;
-                _dataQueue.Add(ByteString.Empty);
```

Comment:
If a blocking take call exists it will be cancelled via cancellation token therefore we don't need to add a dummy anoymore.